### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by zkr024**
* create new classes
* Set up relations between them.
> Belongs to relationships
> Both ways relationships
> Many-to-many relationship

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :label
+
+  def initialize(label)
+    @label = label
+    @students = []
```
Comment (DeliceLydia) [inline classroom.rb:7]: - [x] `students` should be attribute reader

File: classroom.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  end
+
+  def add_student(students)
```
Comment (DeliceLydia) [inline classroom.rb]: - [x] Parameter should be `student` not `students`

File: classroom.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    @students.push(students)
```
Comment (DeliceLydia) [inline classroom.rb]: - [x] Same as here it should be `@students.push(student)`

File: book.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :author, :title
+  attr_reader :rentals
```
Comment (DeliceLydia) [inline book.rb]: - [x] I suggest that you make `rentals` an `attribute_accessor` to be able to access it and modify it

File: book.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (DeliceLydia) [inline book.rb]: - [x] A method for adding `rental` is missing here and remember `book` itself should be sent as a parameter to create the new rental achieving the 'has-many' association.  you can do:
```ruby
 def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: rental.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
```
Comment (DeliceLydia) [inline rental.rb]: - [x] Please create `book`  and `person` attr_accessor to be able to access and modify them at the same time

File: person.rb [hunk 7]
```diff
@@ -2,14 +2,15 @@
...
   def initialize(age, name = 'Unknown', parent_permission: true)
     super()
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
 
   def can_use_services?
```
Comment (DeliceLydia) [inline person.rb]: - [x] A method for adding `rentals` should be present here and remember `person` itself should be sent as a parameter to create the new rental achieving the 'has-many' association you can do:
```ruby
 def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

File: person.rb [hunk 8]
```diff
@@ -2,14 +2,15 @@
 require './decorator'
 class Person < Nameable
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (DeliceLydia) [inline person.rb]: - [x] I suggest that `rentals` be attr_accessor as well to be accessible and editable

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @zkr024, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: person.rb (hunk 8 above)
Comment (zkr024) [reply person.rb]: @DeliceLydia Thank you for your feedback

**Review by keroloussamy — APPROVED** [review]
Hi @zkr024, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To highlight 🎯

- No linter errors.✔️
- Good code design and implementation. ✔️
- Add all the requirements. ✔️

clap



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**